client/ayon_core/plugins/publish/collect_otio_frame_ranges.py

```diff
@@ -1,93 +1,204 @@
-                otio_clip, 0, 0)
-            self.log.debug(
-                ">> retimed_attributes: {}".format(retimed_attributes))
-            media_in = int(retimed_attributes["mediaIn"])
-            media_out = int(retimed_attributes["mediaOut"])
-            frame_end = frame_start + (media_out - media_in) + 1
-            self.log.debug(frame_end)
-
         data = {
             "frameStart": frame_start,
             "frameEnd": frame_end,
             "clipIn": tl_start,
             "clipOut": tl_end - 1,
             "clipInH": tl_start_h,
             "clipOutH": tl_end_h - 1,
+        }
+        instance.data.update(data)
+        self.log.debug(f"Added frame ranges: {pformat(data)}")
+
+
+class CollectOtioSourceRanges(pyblish.api.InstancePlugin):
+    """Collect source media frame ranges from OTIO clip.
+
+    This plugin extracts and stores source media frame ranges including
+    handles from the OTIO clip.
+
+    Requires:
+        otioClip (otio.schema.Clip): OTIO clip object
+
+    Provides:
+        sourceStart (int): Source media start frame
+        sourceEnd (int): Source media end frame
+        sourceStartH (int): Source media start frame with handles
+        sourceEndH (int): Source media end frame with handles
+    """
+
+    label = "Collect Source OTIO Frame Ranges"
+    order = pyblish.api.CollectorOrder - 0.07
+    families = ["shot", "clip"]
+    hosts = ["hiero", "flame"]
+
+    def process(self, instance):
+        """Process the instance to collect source frame ranges.
+
+        Args:
+            instance: The instance to process
+        """
+
+        if not validate_otio_clip(instance, self.log):
+            return
+
+        otio_clip = instance.data["otioClip"]
+
+        # Get source ranges
+        otio_src_range = otio_clip.source_range
+        otio_available_range = otio_clip.available_range()
+        otio_src_range_handles = otio_range_with_handles(otio_src_range, instance)
+
+        # Get source available start frame
+        src_starting_from = otio.opentime.to_frames(
+            otio_available_range.start_time,
+            otio_available_range.start_time.rate
+        )
+
+        # Convert to frames
+        src_start, src_end = otio_range_to_frame_range(otio_src_range)
+        src_start_h, src_end_h = otio_range_to_frame_range(otio_src_range_handles)
+
+        data = {
             "sourceStart": src_starting_from + src_start,
             "sourceEnd": src_starting_from + src_end - 1,
             "sourceStartH": src_starting_from + src_start_h,
             "sourceEndH": src_starting_from + src_end_h - 1,
         }
         instance.data.update(data)
-        self.log.debug(
-            "_ data: {}".format(pformat(data)))
-        self.log.debug(
-            "_ instance.data: {}".format(pformat(instance.data)))
+        self.log.debug(f"Added source ranges: {pformat(data)}")
+
+
+class CollectOtioRetimedRanges(pyblish.api.InstancePlugin):
```

Comment:
There are plugins `CollectOtioFrameRanges`, `CollectOtioSourceRanges`, `CollectOtioRetimedRanges`, is there reason why it's 3 plugins instead of one? From what I see in the code, 1 plugin could be 1 method...

Reply:
Only reason we are splitting it into different plugins is that we need to approach collection of some attributes in particular context, with host and families filtering.

Reply:
`CollectOtioSourceRanges` and `CollectOtioRetimedRanges` have exactly same filters. Hosts filtering overall is related to this comment https://github.com/ynput/ayon-core/pull/1060/files#r1927050138 .

Reply:
implemented in https://github.com/ynput/ayon-core/pull/1060/commits/971c4aef43b5dad3f74b8f8ff321ef66c2927381